packages/core/contracts/data-verification-mechanism/implementation/VotingV2.sol

```diff
@@ -814,48 +795,19 @@ contract VotingV2 is Staker, OracleInterface, OracleAncillaryInterface, OracleGo
-                continue;
-            }
-
-            PriceRequest storage priceRequest = priceRequests[priceRequestIds[requestIndex]];
+            PriceRequest storage priceRequest = priceRequests[resolvedPriceRequestIds[requestIndex]];
             VoteInstance storage voteInstance = priceRequest.voteInstances[priceRequest.lastVotingRound];
 
-            // If the request status is not resolved then: a) Either we are still in the current voting round, in which
-            // case break the loop and stop iterating (all subsequent requests will be in the same state by default) or
-            // b) we have gotten to a rolled vote in which case we need to update some internal trackers for this vote
-            // and set this within the skippedRequestIndexes mapping so the next time we hit this it is skipped.
-            if (!_priceRequestResolved(priceRequest, voteInstance, currentRoundId)) {
-                // If the request is not resolved and the lastVotingRound less than the current round then the vote
-                // must have been rolled. In this case, update the internal trackers for this vote.
-                if (priceRequest.lastVotingRound < currentRoundId) {
-                    priceRequest.lastVotingRound = SafeCast.toUint32(currentRoundId);
-                    priceRequest.priceRequestIndex = SafeCast.toUint64(priceRequestIds.length);
-
-                    // This is a subtle operation. This is not setting the skip value for the _current request_ to this
-                    // value. It is setting the skip value for the element after the last processed index to the skip
-                    // value. This causes this skip interval to extend on each subsequent rolled request because no
-                    // new elements are processed on a skip, thereby leaving nextIndexToProcess the same.
-                    skippedRequestIndexes[nextIndexToProcess] = requestIndex;
-
-                    // Re-enqueue the price request so that it'll be traversed later, when settled and slashing then.
-                    priceRequestIds.push(priceRequestIds[requestIndex]);
-                    continue;
-                }
-                // Else, we are simply evaluating a request that is still actively being voted on. In this case, break.
-                // All subsequent requests within the array must be in the same state and can't have slashing applied.
-                break;
-            }
-
             // If the request we're processing now is not the same round as the last index we processed successfully
             // (not rolled), then we need to apply slashing because there's been a round change.
             if (
                 slash != 0 &&
                 nextIndexToProcess != 0 &&
-                priceRequests[priceRequestIds[nextIndexToProcess - 1]].lastVotingRound != priceRequest.lastVotingRound
+                priceRequests[resolvedPriceRequestIds[nextIndexToProcess - 1]].lastVotingRound !=
```

Comment:
I think you mentioned us being able to simplify these two if statements. I suspect you're right, and I would hope this would make things a lot easier to understand.

By my understanding, whenever we're running this method, we know that all of the previous rounds' resolved requests have been processed, so there's no uncertainty about when the round change will happen.

Please correct me if I'm wrong, but I think the following logic should work:
1. Each time we add a slash, we check if (I think this is similar to the current check here):
    a. There is a next index in the resolved requests array.
    b. That next index (not previous) was resolved in the same round as the current one.
2. If there's an upcoming round change or end of array, then we apply the slash. If not, we add it to unapplied slash.

I would think with that logic, we wouldn't need the second if statement at all, and we'd only have to check this when applying slash. There also may be more elegant ways to check this. Thoughts?

Would something like that work?

Reply:
yup! I've basically implemented this now.